vividus/src/main/java/org/vividus/runner/DeprecatedPropertiesReplacer.java

```diff
@@ -0,0 +1,118 @@
+    public static void main(String[] args) throws IOException, ParseException
+    {
+        Vividus.init();
+
+        Options options = new Options();
+        Option directoryOption = new Option("rl", "resourceLocation", true, "Directory to find properties.");
+        options.addOption(directoryOption);
+        CommandLineParser parser = new DefaultParser();
+        CommandLine commandLine = parser.parse(options, args);
+        String testResourcesLocation = commandLine.getOptionValue(directoryOption.getOpt());
+
+        DeprecatedPropertiesReplacer deprecatedPropertiesReplacer = new DeprecatedPropertiesReplacer(
+                BeanFactory.getResourcePatternResolver());
+        deprecatedPropertiesReplacer.replaceDeprecatedProperties(testResourcesLocation);
+    }
+
+    private void replaceDeprecatedProperties(String testResourcesLocation) throws IOException
+    {
+        loadDeprecatedProperties();
+        Resource[] allPropertiesAsResources = resourcePatternResolver
+                .getResources(Path.of(testResourcesLocation).toUri() + "/**/*.properties");
+
+        for (Resource resource : allPropertiesAsResources)
+        {
+            replaceDeprecatedPropertiesInResource(resource);
+        }
+    }
+
+    private void replaceDeprecatedPropertiesInResource(Resource resource) throws IOException
+    {
+        String propertiesAsString = resource.getContentAsString(StandardCharsets.UTF_8);
+        Properties currentProperties = new Properties();
+        currentProperties.load(new StringReader(propertiesAsString));
+        Properties actualProperties = (Properties) currentProperties.clone();
+
+        DeprecatedPropertiesHandler deprecatedPropertiesHandler = new DeprecatedPropertiesHandler(DEPRECATED_PROPERTIES,
+                PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
+        deprecatedPropertiesHandler.replaceDeprecated(actualProperties);
+
+        for (Map.Entry<Object, Object> entry : currentProperties.entrySet())
+        {
+            String oldKey = (String) entry.getKey();
+            String actualKey = deprecatedPropertiesHandler.getActualPropertyKey(oldKey);
+
+            if (actualKey != null)
+            {
+                propertiesAsString = propertiesAsString.replace(oldKey, actualKey);
+            }
+        }
+        FileUtils.writeStringToFile(resource.getFile(), propertiesAsString, StandardCharsets.UTF_8);
+    }
+
+    private void loadDeprecatedProperties() throws IOException
+    {
+        Resource[] deprecatedPropertiesAsResources = resourcePatternResolver
+                .getResources(DEPRECATED_PROPERTIES_CLASSPATH);
+        for (Resource deprecatedPropertiesFile : deprecatedPropertiesAsResources)
+        {
+            Properties deprecatedProperties = new Properties();
+            deprecatedProperties
+                    .load(new StringReader(deprecatedPropertiesFile.getContentAsString(StandardCharsets.UTF_8)));
+            DEPRECATED_PROPERTIES.putAll(deprecatedProperties);
```

Comment:
can we do this as in ConfigurationResolver?
propertiesLoader.loadFromResourceTreeRecursively(false, "deprecated");

Reply:
> loadFromResourceTreeRecursively

In this case we need to open access to `PropertiesLoader` class and necessary method
I'm not sure this strictly necessary (same about recursive loading)